Implement RpcMessage `AccessLevel` new meta-attribute

j4r0u53k opened the pull request.

j4r0u53k committed:
Add `AccessLevel` meta-attribute for RpcMessage

A new attribute to specify access level in requests by a numerical
representation.

The older `Access` meta-attribute, which uses a string representation
is kept for compatibility with older brokers/clients as a fallback
if `AccessLevel` is not present.

The `Access` to `AccessLevel` conversion is performed as follows:

If the `Access` string contains a comma-separated list of grants
the first known grant that a conversion from string succeeds for
is returned, the rest is ignored.

(#14)

j4r0u53k committed:
Rename enum `Access` to `AccessLevel`

The new name matches with the new meta-attribute.

RpcMessage methods `access` and `set_access` renamed
appropriately.

(#14)

j4r0u53k committed:
RpcMessage: add tests for AccessLevel

(#14)

fvacek commented on src/metamethod.rs:
```diff
@@ -26,26 +27,53 @@ impl From<u8> for Flag {
+            "dev" => Some(AccessLevel::Developer),
+            "su" => Some(AccessLevel::Superuser),
+            // For backward compatibility with the dot-local-hack
+            self::DOT_LOCAL_GRANT => Some(AccessLevel::Superuser),
```
`.local` should return `None`

j4r0u53k replied:
Fixed

j4r0u53k committed:
Provide backward compatibility for the dot-local hack

Broker sets both 'Access' and 'AccessLevel' RpcMessage meta
attributes or pass them as is from a parent broker in order
to keep backward compatibility with older brokers that read
'dot-local' in the 'Access' string to expose their '.local'
dir.

The AccessLevel is treated as plain i32, so broker can evaluate
access also for values that are not defined in the enum.

(#14)
